packages/rum-core/src/common/url.js

```diff
@@ -136,6 +136,24 @@ class Url {
 
     this.protocol = protocol || location.protocol
 
+    /**
+     * Construct port and hostname from host and assign defaults based
+     * on http and https protocols
+     */
+    if (/:\d+$/.test(this.host)) {
+      const value = this.host.split(':')
+      this.port = value.pop()
+      this.hostname = value.join(':')
+    } else {
+      this.hostname = this.host
+      this.port =
+        this.protocol === 'http:'
+          ? '80'
+          : this.protocol === 'https:'
+          ? '443'
+          : ''
+    }
```

Comment:
URL native instrumentation does not set `port` for default ports and it makes there.But for us, It would make out service and address construction rely on additional checks.  

IPv6 parsing is on par with browser URL implementation and I have added comment in context.js on reasoning 
<img width="258" alt="Screenshot 2019-12-10 at 12 18 06" src="https://user-images.githubusercontent.com/3902525/70525269-3db32500-1b47-11ea-8124-a23332b76c1a.png">

Reply:
Regarding the default port, we can set it explicitly for the service map.